src/components/Form/BackButton/BackButton.js

```diff
@@ -0,0 +1,50 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+// Import custom hooks
+import useStepLogic from '../useStepLogic';
+
+const BackButton = ({ setIsFormStarted }) => {
+  const { formDataState, setStep } = useStepLogic(); // Custom hook for handling continue button (validation, errors etc)
+
+  const { currentStep, CustomerType, Application } = formDataState;
+
+  const goBack = () => {
+    if (currentStep > 1) {
+      if (
+        currentStep === 4 &&
+        (CustomerType === 'Scratchcard' || CustomerType === 'ClassPass')
+      ) {
+        setStep(currentStep - 3);
+      } else if (
+        currentStep === 3 &&
+        CustomerType === 'SwiftPortal' &&
+        !Application.CustomerTypeStep2
+      ) {
+        setStep(currentStep - 2);
+      } else {
+        setStep(currentStep - 1);
+      }
+    } else {
+      setStep(1);
+      setIsFormStarted(false);
+    }
+  };
```

Comment:
This isn't working as expected.
To replicate the bug:

1. Click Start
2. Click "Swift Card" in step 1
3. Click continue
4. Press back button
5. Press back button again
6. You should now be back on the intro page
7. Click "call us" link under "before you start" section
8. Click back button
9. Bug

Reply:
Confirmed as working